internal/git/gitsync_processor.go

```diff
@@ -123,14 +130,13 @@ func watchRepo(ctx context.Context, repo *v1.RepositoryPath, _ /* namespace */ s
 	return nil
 }
 
-func NewGitSyncProcessor(ctx context.Context, gitSync *v1.GitSync, k8client client.Client, clusterName string) (*GitSyncProcessor, error) {
+func NewGitSyncProcessor(ctx context.Context, gitSync *v1.GitSync, config *rest.Config, clusterName string) (*GitSyncProcessor, error) {
```

Comment:
I'm not sure, but I wonder if we still want to have the `k8client client.Client` in there or not? (which is I guess for `GitSyncs` in particular) Or alternatively the Status subresource used [here](https://github.com/numaproj-labs/numaplane/blob/main/internal/controller/gitsync_controller.go#L127)? We also need to update the `Status.CommitStatus`. Or did you imagine we would derive a client from the config for that as well?

Reply:
I'm thinking to have one client which we can derive from rest config but I'm open for both as well, if we want to keep k8sclient also I'll add that?

Reply:
I see it's nice to eliminate redundancy - but it does also save some steps not to have to derive the client. 
One thing is: I am now thinking that `k8client client.Client` and `config *rest.Config` could use more meaningful names. 

I think `k8client` may only be for numaplane CRDs? But I haven't tested to verify it. If so, maybe we should have a name that reflects that?

And then `config` could use a more meaningful name as well. 

What do you think? 

Also, by the way, if you do prefer not to have the redundancy of the two parameters, can you create the client for the subresource so it can be used by @xdevxy 's PR?

Reply:
I'm thinking to refactor it a bit like from main file itself I'll generate this kubernetes.client and pass it all over where ever it required and so that I don't have to pass restConfig as well as k8sClient all over and also I have a method inside client interface defined like Get, Delete and others? probably we can discuss more over call?

For now I'm thinking to not remove the k8sClient from here and keep both because refactor PR will be a bit longer?